userspace/libscap/scap.c

```diff
@@ -1464,6 +1465,57 @@ uint64_t scap_get_driver_schema_version(scap_t* handle)
 	return 0;
 }
 
+void scap_gethostname(scap_t* handle)
+{
+	gethostname(handle->m_machine_info.hostname, sizeof(handle->m_machine_info.hostname) / sizeof(handle->m_machine_info.hostname[0]));
+	char *env_hostname;
+	env_hostname = getenv("FALCO_HOSTNAME");
+	if(env_hostname != NULL)
+	{
+		snprintf(handle->m_machine_info.hostname, sizeof(handle->m_machine_info.hostname), "%s", env_hostname);
+	}
+}
+
+void scap_get_self_pid_start_ts(scap_t* handle)
+{
+	struct stat st = {0};
+	char path[256];
+	// proc dir as seen from the calling process pid namespace
+	snprintf(path, sizeof(path), "/proc/%d/", getpid());
+	if(stat(path, &st) == 0)
+	{
+		handle->m_machine_info.self_pid_start_ts = st.st_ctim.tv_sec * (uint64_t) 1000000000 + st.st_ctim.tv_nsec;
+	}
+}
+
+void scap_get_bpf_stats_enabled(scap_t* handle)
+{
+	FILE* f;
+	// proc dir as seen from the calling process pid namespace
+	if((f = fopen("/proc/sys/kernel/bpf_stats_enabled", "r")))
```

Comment:
This will be used to be able to activate `bpftool` like stats as we discussed here https://github.com/falcosecurity/falco/issues/2222#issuecomment-1312223538 and also seems a useful info to know about a machine as enabling bpf stats can add a slight performance overhead as stated in the docs https://manpages.ubuntu.com/manpages/focal/en/man8/bpftool-prog.8.html.